### DISCUSSION TITLE
I am doing project called SDN effect on smart grid networks , while build the project I getting this error

### DISCUSSION THREAD

**Discussion by thahseeenaabdulrassak** [omnetpp/omnetpp#1096] [Q&A]
11:34:43 **** Incremental Build of configuration release for project SDN-Smart-Grid ****
make MODE=release all 
cd src && /usr/bin/make
make[1]: Entering directory '/c/Users/thahz/OneDrive/Desktop/Projekt/RP_Projekt/SDN-Smart-Grid/SDN-Smart-Grid/src'
Creating executable: ../out/clang-release/src/SDN-Smart-Grid.exe
lld: error: unable to find library -lopenflow6
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [Makefile:104: ../out/clang-release/src/SDN-Smart-Grid.exe] Error 1
make[1]: Leaving directory '/c/Users/thahz/OneDrive/Desktop/Projekt/RP_Projekt/SDN-Smart-Grid/SDN-Smart-Grid/src'
make: *** [Makefile:2: all] Error 2
"make MODE=release all" terminated with exit code 2. Build might be incomplete.

11:34:44 Build Failed. 2 errors, 0 warnings. (took 1s.224ms)

**Comment by rhornig** [comment]
The project producing the openflow6 libraray must be successfully built and the library path provided by for your project in the linker phase mist be secified (-L /path/to/OPENFLOW/library)

@opp-assistant What is your take on this?

**Comment by opp-assistant** [comment]
The error you are encountering is a linker error, specifically indicating that the linker (`lld`) is unable to find the library `-lopenflow6`. This is typically due to one of the following reasons:

1. **Library Not Installed:** The library `openflow6` might not be installed on your system. Ensure that this library is installed. If it's a custom or third-party library, you might need to download and install it manually.

2. **Incorrect Library Path:** If the library is installed, the linker might not be able to find it due to an incorrect library path. You need to specify the correct path to the library in your project's linker settings. This can be done by modifying the linker command or the makefile to include the correct path to the library.

3. **Library Naming Issue:** Sometimes, the issue might be due to a mismatch in the library name. For instance, the library might be named differently on your system. Double-check the actual name of the library file on your system.

4. **Environment Configuration:** In some cases, environment variables like `LD_LIBRARY_PATH` (on Unix-like systems) need to be correctly set so that the linker knows where to look for the libraries.

To resolve this issue, you should:

- Verify that the `openflow6` library is indeed installed on your system.
- If installed, ensure that the path to the library is correctly specified in your project's settings or makefile.
- Check the actual name of the library file on your system to ensure it matches what's being referenced in your build process.
- Optionally, set or modify the appropriate environment variables to help the linker locate the library.

Since you are working within the context of an OMNeT++ project, also make sure that any specific dependencies or settings required by OMNeT++ for this library are correctly configured. This can include checking the OMNeT++ project settings and ensuring that any OMNeT++ specific environment variables or configurations are correctly set.
